lesson-5-1

katemikh opened the pull request:
Homework Assignment - lesson-5-1

katemikh committed:
lesson-5-1

mgj2 commented:
Hey Kateryna,
From what I can tell using an HTML previewer, you've still got some bugs to work out here.  This lesson is definitely different from what we've been focusing on, so far, so please reach out if you'd like any assistance!

katemikh committed:
corrected code lesson-5-1

katemikh commented:
please find correction @mgj2  !

1) line 105: maxNumberOfAttempts = 5;
I removed it completely, because... on line 15 has already been declared and assigned
 
2) line 108 (was 109):
submitButton.disabeld = false;
fixed disabeld to disabled
 
3) line 35-37:
added code
  if (!guess) {
    return;
  }
 
so that if the user simply presses an empty button, nothing happens
 
4) removed lines 47 and 54 - guessInput.value = '';
 and moved it to line 41 (so as not to repeat it several times)
 
5) corrected the logic 46-49 (if less than 1) and 51-54 (if more than 100). Show a message that already exists.
It became like this
  if (guess <= 0) {
    belowZeroMessage.style.display = 'block';
    return;
  }
 
  if (guess > 99) {
    aboveHundredMessage.style.display = 'block';
    return;
  }
 
6) Moved the red highlight from point 5 to css (added lines 17-20 to css)
    #below-zero,
    #above-hundred {
          color: red;
}

7) moved line 42 attempts = attempts + 1; to line 55.
So that the checks are performed first, and then the attempt is counted. If the test fails, the attempt is not counted.
 
8) The code about attempts from lines 58, 59 and 77-80 was moved above outside the blocks (to lines 57-63) into a separate function. Because if you guess correctly on the first try, you will have 'You made 1 gesses'. To avoid repeating this code twice.
 
9) added line 74 - return. If you guess correctly, that’s it, further execution of the function stops.
